python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -644,7 +644,8 @@ def replicate(self, destSE, sourceSE=''):
         logger.info("Replicating file %s to %s" % (self.lfn, destSE))
         stdout = execute('replicateFile("%s", "%s", "%s")' % (self.lfn, destSE, sourceSE))
         if isinstance(stdout, dict) and stdout.get('OK', False) and self.lfn in stdout.get('Value', {'Successful': {}})['Successful']:
-            self.locations.append(destSE)
+            if destSE not in self.locations:
```

Comment:
But this is a rather long winded way of getting here than ```if destSE in self.locations: return```, if it's harmless then ok, I suppose in some situation the locations may be out of date enough to reflect where a true lfn replica is